### PR TITLE
IDGEN-77 LogEntry Resource Layout

### PR DESCRIPTION
**Pull request by cnagadya**
## JIRA TICKET NAME:

[IDGEN-77 Create Log Entries Resource](https://issues.openmrs.org/browse/IDGEN-77)

## SUMMARY:
Create Resource for the Log Entries endpoint that searches through and lists all ID Generation activities on OpenMRS

_Initial Commit: Only the class set up_

### CODE REVIEW

**Comment by dkayiwa** [comment]
@cnagadya this is a very good start! 👍 
Which existing search handler did you copy from?

**Comment by cnagadya** [comment]
@dkayiwa, in the step by step guide, the [ConceptSearchHandler1_8](https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java) was mentioned as an example that had implemented a custom search so I used it as a reference

**Comment by dkayiwa** [comment]
@cnagadya did you intentionally make your base class different from that of ConceptSearchHandler1_8?

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package org.openmrs.module.idgen.rest.resource;
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: You will also need to add the openmrs license header file.

**Comment by cnagadya** [comment]
@dkayiwa thought the filename and base class name should be the same. Regarding the methods, the additional methods are for the DelegatingCrudResource superclass which according to the [step-by-step guide](https://wiki.openmrs.org/display/docs/Adding+a+Web+Service+Step+by+Step+Guide+for+Core+Developers) could be used for object types other than meta data. Dont know whether the LogEntry class needed to extend any of the helper classes though and I had raised a question about this on IRC

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java (hunk 1 above)
Comment (cnagadya) [reply omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: Fixed this. Should reflect on the next commit :)

**Comment by dkayiwa** [comment]
@cnagadya did you take a look at the base classes of the search handlers in the folder link i gave in the ticket description?

**Comment by cnagadya** [comment]
Just going through the links now. I am yet to get to the search handlers section

**Comment by dkayiwa** [comment]
And if you get blocked, do not wait for scrum time. ☺️

**Comment by cnagadya** [comment]
Alright, will definitely reach out as soon as I hit a stall

**Comment by cnagadya** [comment]
@dkayiwa, the Search Criteria Builder for Log Entries doesnt exist
how do I go about creating the getPageableResult method in the log entry resource?

**Comment by dkayiwa** [comment]
@cnagadya where does that method come from?

**Comment by cnagadya** [comment]
I dont know whether I got this right but according to the documentation, PageableResult = Helper interface used by implementations that want to do pagination which I believe would case for the doGetAll and doSearch for log entries which means they'll return getPageableResult(context) as in the [example](https://github.com/openmrs/openmrs-module-metadatamapping/blob/master/omod/src/main/java/org/openmrs/module/metadatamapping/web/rest/MetadataTermMappingResource.java). When I use that approach, I'll need to define the getPageableResult method

**Comment by dkayiwa** [comment]
@cnagadya did you take a look at the search handlers whose link i gave in the ticket description?

**Pull request closed without merging**